numpy/core/src/multiarray/arraytypes.c.src

```diff
@@ -3692,9 +3691,12 @@ static void
         }
     }
     else {
-        for (i = 0; i < ni; i++) {
+        for (i = 0, j = 0; i < ni; i++, j++) {
+            if (j >= nv) {
+                j = 0;
+            }
             if (mask[i]) {
-                in[i] = vals[i%nv];
+                in[i] = vals[i];
```

Comment:
Ooops, that's embarrassing... While update the benchmarks, and take a look at expanding the tests.